### PR TITLE
CheckPoint FW v2 - moved methods into client

### PR DESCRIPTION
**Pull request by dorschw**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39704

## Description
Moved non-method clients into the client, refactoring.

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py [hunk 1]
```diff
@@ -16,23 +13,68 @@ class Client(BaseClient):
...
+
+        json_body = {'user': username, 'password': password, 'session-timeout': session_timeout}
+        if domain_arg:
+            json_body['domain'] = domain_arg
+
+        response = self._http_request(method='POST', url_suffix='login', json_data=json_body, headers=self.headers)
+        sid = response.get('sid', '')
+
+        if sid:
+            self.sid = sid
```
Comment (dorschw) [inline Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:43]: @DeanArbel should we set it even if it's `None`/`''` etc?

Comment (DeanArbel) [reply Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2.py:43]: Leave it as is. It was an unsuccessful login, so no need to remove an (potential) existing session, that's (could be) stored in `self.sid`.

**Review by DeanArbel — APPROVED** [review]

**Comment by DeanArbel** [comment]
@amshamah419 would you mind reviewing the RN for this?

File: Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### CheckPoint Firewall v2
+- Moved non-`Client` methods into the client, for better proxy support. 
+- Better error handling
```
Comment (roysagi) [inline Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md:5]: @dorschw - These notes are more dev-oriented. The notes should be more aligned with what the customer is experiencing. Try to think how they might experience the change made - you can look at some best practices here: https://xsoar.pan.dev/docs/documentation/release-notes#examples-and-best-practices

Comment (roysagi) [reply Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md:5]: Also stay aligned with the formats in the docs please

Comment (dorschw) [reply Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md]: done, can you take another look please?

File: Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### CheckPoint Firewall v2
+- Fixed issues with logging in and calling commands via proxy connection.
+- Improved error handling.
```
Comment (DeanArbel) [inline Packs/CheckpointFirewall/ReleaseNotes/2_1_0.md:5]:
```suggestion
##### CheckPoint Firewall v2
- Fixed issues when logging in and using the integration with **proxy** connection.
```

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**